Source/Core/VideoCommon/RenderState.cpp

```diff
@@ -202,31 +203,69 @@ void BlendingState::ApproximateLogicOpWithBlending()
-      {false, SrcBlendFactor::InvDstClr, DstBlendFactor::Zero},       // EQUIV
-      {false, SrcBlendFactor::InvDstClr, DstBlendFactor::InvSrcClr},  // INVERT
-      {false, SrcBlendFactor::One, DstBlendFactor::InvDstAlpha},      // OR_REVERSE
-      {false, SrcBlendFactor::InvDstClr, DstBlendFactor::InvSrcClr},  // COPY_INVERTED
-      {false, SrcBlendFactor::InvDstClr, DstBlendFactor::One},        // OR_INVERTED
-      {false, SrcBlendFactor::InvDstClr, DstBlendFactor::InvSrcClr},  // NAND
-      {false, SrcBlendFactor::One, DstBlendFactor::One},              // SET
+      // clang-format off
+      {false, false, SrcBlendFactor::One,       DstBlendFactor::Zero},        // CLEAR (Shader outputs 0)
+      {true,  false, SrcBlendFactor::DstClr,    DstBlendFactor::Zero},        // AND
+      {true,  true,  SrcBlendFactor::One,       DstBlendFactor::InvSrcClr},   // AND_REVERSE
+      {false, false, SrcBlendFactor::One,       DstBlendFactor::Zero},        // COPY
+      {true,  true,  SrcBlendFactor::DstClr,    DstBlendFactor::One},         // AND_INVERTED
+      {true,  false, SrcBlendFactor::Zero,      DstBlendFactor::One},         // NOOP
+      {true,  false, SrcBlendFactor::InvDstClr, DstBlendFactor::InvSrcClr},   // XOR
+      {true,  false, SrcBlendFactor::InvDstClr, DstBlendFactor::One},         // OR
+      {true,  false, SrcBlendFactor::InvDstClr, DstBlendFactor::InvSrcClr},   // NOR
+      {true,  false, SrcBlendFactor::InvDstClr, DstBlendFactor::Zero},        // EQUIV
+      {true,  false, SrcBlendFactor::InvDstClr, DstBlendFactor::Zero},        // INVERT (Shader outputs 255)
+      {true,  false, SrcBlendFactor::One,       DstBlendFactor::InvDstAlpha}, // OR_REVERSE
+      {false, false, SrcBlendFactor::One,       DstBlendFactor::Zero},        // COPY_INVERTED (Shader inverts)
+      {true,  false, SrcBlendFactor::InvDstClr, DstBlendFactor::One},         // OR_INVERTED
+      {true,  false, SrcBlendFactor::InvDstClr, DstBlendFactor::InvSrcClr},   // NAND
+      {false, false, SrcBlendFactor::One,       DstBlendFactor::Zero},        // SET (Shader outputs 255)
+      // clang-format on
   }};
 
   logicopenable = false;
-  blendenable = true;
-  subtract = approximations[u32(logicmode.Value())].subtract;
-  srcfactor = approximations[u32(logicmode.Value())].srcfactor;
-  srcfactoralpha = approximations[u32(logicmode.Value())].srcfactor;
-  dstfactor = approximations[u32(logicmode.Value())].dstfactor;
-  dstfactoralpha = approximations[u32(logicmode.Value())].dstfactor;
+  usedualsrc = false;
+  const LogicOpApproximation& approximation = approximations[static_cast<u32>(logicmode.Value())];
+  if (approximation.blendEnable)
+  {
+    blendenable = true;
+    subtract = approximation.subtract;
+    srcfactor = approximation.srcfactor;
+    srcfactoralpha = approximation.srcfactor;
+    dstfactor = approximation.dstfactor;
+    dstfactoralpha = approximation.dstfactor;
+  }
+}
+
+bool BlendingState::LogicOpApproximationIsExact()
```

Comment:
This is minor but if you keep this function and make the changes suggested, I'd also recommend changing the name to `ShouldApproximateLogicOpWithBlending()`.